tests/rust/assoc_types.rs

```diff
@@ -0,0 +1,41 @@
+
+pub const PI: f64 = 3.1415926535;
+pub const MAX: <MyStruct as MyTrait>::SomeType = 50;
+
+pub trait MyTrait{
+    type SomeType;
+}
+
+#[repr(C)]
+pub struct MyStruct{
+    a: u8,
+}
+
+
+impl MyTrait for MyStruct{
+    type SomeType = i64;
+}
+
+
+#[repr(C)]
+pub struct AnotherStruct{
+    a: u8,
+    b: <MyStruct as MyTrait>::SomeType,
+}
+
+#[no_mangle]
+pub extern fn another_struct(struct_: &AnotherStruct){}
+
+#[no_mangle]
+pub extern fn test_fn(struct_: &MyStruct) -> <MyStruct as MyTrait>::SomeType {
+    0
+}
```

Comment:
```suggestion
pub extern fn test_fn(struct_: &<MyStruct as MyTrait>::SomeType) -> &<MyStruct as MyTrait>::SomeType {
    struct_
}
```

can we do it like this?